### PR TITLE
Store static images (profile pictures, park pictures) with relative path in database

### LINKED ISSUE #189: Add static park images

### LINKED ISSUE #202: Store static images (profile pictures, park pictures) with relative path in database
Currently profile pictures are only being identified by name

### PR DESCRIPTION
**Pull request by b3h3m0th**
| Q                    | A                                                  |
| -------------------- | -------------------------------------------------- |
| Bug fix?             | no |
| New feature?         | no |
| Related issues/PRs   | closes #189, closes #202  |
| External ticket link |      |
| License              | MIT                                                |

### CODE REVIEW

File: libs/server/common/src/lib/interceptors/map-park-image-path.interceptor.ts [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+import deepdash from 'deepdash';
+const _ = deepdash(lodash);
+
+@Injectable()
+export class MapParkImagePathInterceptor implements NestInterceptor {
+  private keysToBeMapped: (string | number)[] = ['image'];
+
+  private mapImagePath<T = any>(value: T): T {
+    const mapped = _.mapValuesDeep(value, (value, key) => {
+      if (!this.keysToBeMapped.includes(key) || _.isNil(value)) return value;
```
Comment (TimonStadelmann) [inline libs/server/common/src/lib/interceptors/map-park-image-path.interceptor.ts:18]: was macht .isNil

File: libs/server/controllers/user/src/lib/user.service.ts [hunk 2]
```diff
@@ -235,7 +235,7 @@ export class UserService {
         id,
       },
       data: {
-        profilePicture: `${file.filename}`,
+        profilePicture: `${MulterService.destinations.profilePictures}/${file.filename}`,
```
Comment (TimonStadelmann) [inline libs/server/controllers/user/src/lib/user.service.ts:238]: gibt MulterService.destinations.profilePicture da ganze path zrück oder nur /profilepicture oder so ?

File: prisma/seed.prod.ts [hunk 3]
```diff
@@ -10,7 +10,7 @@ const updateParks = async () => {
       id: 1,
       name: 'Generationen Park Mariahilf',
       address: 'Mariahilfstraße 50, 6900 Bregenz',
-      image: 'https://picsum.photos/400/200',
+      image: 'parks/images/generationen-park-mariahilf.png',
```
Comment (TimonStadelmann) [inline prisma/seed.prod.ts:13]: für was der images ordner dazwischen?

**Review by TimonStadelmann — APPROVED** [review]

File: libs/server/common/src/lib/interceptors/map-park-image-path.interceptor.ts (hunk 1 above)
Comment (b3h3m0th) [reply libs/server/common/src/lib/interceptors/map-park-image-path.interceptor.ts:18]: des checkt quasi (`value !== null && value !== undefined)`

File: libs/server/controllers/user/src/lib/user.service.ts (hunk 2 above)
Comment (b3h3m0th) [reply libs/server/controllers/user/src/lib/user.service.ts:238]: beispiel für an entry in da datenbank: `"uploads/profile-pictures/file.jpg"`

File: prisma/seed.prod.ts (hunk 3 above)
Comment (b3h3m0th) [reply prisma/seed.prod.ts:13]: Kompromiss für static files:

- `"static/parks/file.jpg"`
- `"static/exercises/file.mp4"`

**Review by TimonStadelmann — APPROVED** [review]

**Pull request merged**